### ISSUE TITLE
Ansible Lint Github Action ansible-lint-requirements.txt\n or **/pyproject.toml not found

### ISSUE THREAD

**Issue by willbrownecs** [ansible/ansible-lint#4341]
##### Summary

Ansible-lint action errors out with `No file in /home/runner/work/<redacted>/<redacted> matched to [/home/runner/work/<redacted>/<redacted>/.git/ansible-lint-requirements.txt\n or **/pyproject.toml], make sure you have checked out the target repository`

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ubuntu-latest
```

- ansible installation method: github action
- ansible-lint installation method: github action

##### STEPS TO REPRODUCE

```console (paste below)
---
# .github/workflows/ansible-lint.yml
# https://github.com/marketplace/actions/run-ansible-lint
name: ansible-lint
on:
  pull_request:
    branches: [main, stable, release/v*]
jobs:
  build:
    name: Ansible Lint # Naming the build is important to use it as a status check
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - name: Run ansible-lint
        uses: ansible/ansible-lint@main # or version tag instead of 'main'
```
specifically at time of issue the current commit hash of latest/main is 4e3058ffb438604d377ca25fc8c09df765844067

##### Desired Behavior
ansible-lint executes properly

Possible security bugs should be reported via email to `security@ansible.com`

##### Actual Behavior

Please give some details of what is happening. Include a [minimum complete
verifiable example] with:

github action fails to execute with the following error: `No file in /home/runner/work/<redacted>/<redacted> matched to [/home/runner/work/<redacted>/<redacted>/.git/ansible-lint-requirements.txt\n or **/pyproject.toml], make sure you have checked out the target repository`

Reverting the `uses` block from `uses: ansible/ansible-lint@main` to `uses: ansible/ansible-lint@571711ffc3036d6893213fbe23d68f9d73b983ce` resolves the issue. This commit (at time of writing) is one commit behind main/latest.

[minimum complete verifiable example]: http://stackoverflow.com/help/mcve

**Comment by sdodsley** [comment]
I'm seeing this as well. Only started happening in the last few hours

**Comment by shaunmcevoy** [comment]
Causing issues here as well.   Thanks for workaround @willbrownecs

**Comment by ssbarnea** [comment]
I do not see any links to a repository with this failure so I am unable to reproduce/debug this failure.

**Comment by willbrownecs** [comment]
> I do not see any links to a repository with this failure so I am unable to reproduce/debug this failure.

I am unfortunately unable to link to my repository as it is internal. Maybe one of the other kind folks with the same issue can?

**Comment by ssbarnea** [comment]
> > I do not see any links to a repository with this failure so I am unable to reproduce/debug this failure.
> 
> I am unfortunately unable to link to my repository as it is internal. Maybe one of the other kind folks with the same issue can?

At least try to debug it by looking at the logs. It does not do so many things so it should be easy to spot the root cause.

**Comment by sdodsley** [comment]
https://github.com/Pure-Storage-Ansible/FlashArray-Collection/actions/runs/10944918604/job/30389000669

**Comment by ssbarnea** [comment]
@sdodsley Thanks! #4342 should fix it. What I do not understand is why the test suite passed, that is a real dillema.

**Comment by ssbarnea** [comment]
I just merged the fix, I need confirmation from someone using `@main` that current code is working fine, so I can also tag a new release.

**Comment by sdodsley** [comment]
Confirmed the fix is good - working now.
